tests/integration/README.md

```diff
@@ -0,0 +1,219 @@
+
+```bash
+# Show help message for the check command.
+$ python3 ./tests/integration/run.py check -h
+```
+
+
+## Test Case Configuration
+
+The utility interacts with integration test cases. Each test case locates in a single directory containing a test config file `test.yaml`, alongside other files required for the test case (e.g. config files, policy files, provenance expectation files).
+
+Here is an example. The test case directory looks like this. Alongside the `test.yaml` file, there is a file named `analysis_report.json` storing the expected output of an analysis report.
+
+```
+./tests/integration/cases/apache_maven/
+├── analysis_report.json
+└── test.yaml
+```
+
+Here is the content of the `test.yaml` file.
+
+```yaml
+description: >
+  Analyze with PURL and repository path without dependency resolution.
+
+steps:
+- name: Run macaron analyze
+  kind: analyze
+  options:
+    command_args:
+    - --package-url
+    - pkg:maven/apache/maven
+    - --repo-path
+    - https://github.com/apache/maven
+    - -b
+    - master
+    - -d
+    - 3fc399318edef0d5ba593723a24fff64291d6f9b
+    - --skip-deps
+- name: Compare analysis report
+  kind: compare
+  options:
+    kind: analysis_report
+    result: output/reports/maven/apache/maven/maven.json
+    expected: analysis_report.json
+```
+
+Each test config file has a description at the top telling what the test case is about, and a sequence of steps to be executed. The execution of a test case stops whenever a step fails (i.e. exits with non-zero code, by default). In the example above, there are 2 steps: (1) run the `macaron analyze` command, and (2) compare a JSON analysis report with the corresponding expected result file. Note that all steps are executed with the test case directory being the current working directory. Therefore, all filepaths in the test config file are relative to the test case directory.
+
+
+## Example Usages
+
+### Create a new test case
+
+You create a new test case by creating a new directory, then a `test.yaml` within it. To schema-validate the `test.yaml` file, you can use the `check` command and point to the test case directory:
+
+```bash
+# Schema-validate the ./test/case/directory/test.yaml file.
+$ python3 ./tests/integration/run.py check ./test/case/directory
+```
+
+At this point, some expected result files does not exist yet, since you normally want to run `macaron` once, inspect the result files, then turn them to expected result files if they look good. To do this, you can run in **interactive** mode. In this mode, the utility stops at each steps and ask if you want to run or skip a step. For compare steps, the utility also ask if you want to "update" the expected result file instead of compare.
```

Comment:
nit:
```suggestion
At this point, some expected result files does not exist yet, since you normally want to run `macaron` once, inspect the result files, then turn them to expected result files if they look good. To do this, you can run in **interactive** mode. In this mode, the utility stops at each steps and ask if you want to run or skip a step. For `compare` steps, the utility also ask if you want to "update" the expected result file instead of compare.
```

Reply:
```suggestion
At this point, some expected result files do not exist yet, since you normally want to run `macaron` once, inspect the result files, then turn them into expected result files if they look good enough. To do this, you can run in **interactive** mode. In this mode, the utility stops at each step and asks if you want to run or skip a step. For `compare` steps, the utility also asks if you want to "update" the expected result file instead of compare.
```

Reply:
Thanks a lot. Fixed in 0a816d8dd918b77837b5c4e48cde4f689a37b630.